crates/turbopack-tests/tests/snapshot.rs

```diff
@@ -244,12 +242,39 @@ fn remove_file(root: &str, path: &str) -> Result<()> {
     Ok(())
 }
 
+/// Removes annoying hash fingerprints from files paths.
+///
+/// If the hash changes whenever the contents of the file changes, then git will
+/// show it as a brand new file instead of a diff of an exsting file. This makes
+/// reviewing changes extremely difficult.
+async fn remove_hash_fingerprint(asset: AssetVc) -> Result<FileSystemPathVc> {
+    let path = asset.path();
+
+    if EcmascriptChunkSourceMapAssetVc::resolve_from(asset)
+        .await?
+        .is_some()
+    {
+        // .map files have a hash like foo.js.abc123.map.
+        let mut p = PathBuf::from(&path.await?.path);
+        if p.extension() == Some(OsStr::new("map")) {
+            p.set_extension("");
+            debug_assert_ne!(p.extension(), Some(OsStr::new("js")));
```

Comment:
In what case is this possible, since we set the extension to "" just before?

Reply:
It recomputes afterwards, inheriting the next `.abc123` that still remains. I think it's implemented like an `rfind`.